### PR TITLE
PHOENIX-5173: LIKE and ILIKE statements return empty result list for …

### PR DESCRIPTION
**Pull request by swaroopak**
…search without wildcard

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/LikeExpressionIT.java [hunk 1]
```diff
@@ -430,4 +430,26 @@ public void testParameterizedLikeExpression() throws Exception {
         rs = select.executeQuery();
         assertFalse(rs.next());
     }
+
+    @Test
+    public void testLikeExpression5173() throws Exception {
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/LikeExpressionIT.java]: please rename the test and instead add a comment pointing towards the JIRA instead.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/LikeExpressionIT.java [hunk 2]
```diff
@@ -430,4 +430,26 @@ public void testParameterizedLikeExpression() throws Exception {
...
+            final String upsertTable = "UPSERT INTO " + table + " VALUES(1, 'Some Name')";
+            conn.createStatement().executeUpdate(upsertTable);
+            String likeSelect = "SELECT * FROM " + table + " WHERE USER_NAME LIKE 'Some Name'";
+            ResultSet rs = conn.createStatement().executeQuery(likeSelect);
+            assertTrue(rs.next());
+            assertFalse(rs.next());
+            String iLikeSelect = "SELECT * FROM " + table + " WHERE USER_NAME ILIKE 'soMe nAme'";
+            rs = conn.createStatement().executeQuery(iLikeSelect);
+            assertTrue(rs.next());
+            assertFalse(rs.next());
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/LikeExpressionIT.java]: nit: close the `ResultSet`

File: phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java [hunk 3]
```diff
@@ -507,9 +507,6 @@ public Expression visitLeave(LikeParseNode node, List<Expression> children) thro
             }
             if (index == -1) {
                 String rhsLiteral = LikeExpression.unescapeLike(pattern);
-                if (lhsMaxLength != null && lhsMaxLength != rhsLiteral.length()) {
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java]: Looks good, but just confirm that this does cause any regressions to existing like conditions that have wildcards?

**Comment by twdsilva** [comment]
+1

File: phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java (hunk 3 above)
Comment (swaroopak) [reply phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java]: Yes. :)

**Comment by swaroopak** [comment]
@ChinmaySKulkarni Could you please commit this one? I addressed the comments.

**Pull request closed without merging**